activerecord/test/cases/query_cache_test.rb

```diff
@@ -323,17 +322,16 @@ def test_query_cache_across_threads
 
             ActiveRecord::Base.connection_handler.clear_active_connections!(:all)
           }.call({})
+
+          assert_cache :off, thread_2_connection
         }
 
         started.wait
 
         thread_1_connection = ActiveRecord::Base.lease_connection
         assert_not_equal thread_1_connection, thread_2_connection
-        assert_cache :dirty, thread_2_connection
```

Comment:
We can no longer assert on the cache store of `thread_2_connection` outside of thread 2 because this would point to a separate cache store.

We have `assert_cache :dirty` within the thread anyway so I think that should be enough.